### DISCUSSION TITLE
ENOSPC no space left on disk write

### DISCUSSION THREAD

**Discussion by TheWomanInYellow** [cat-box/aws-foundry-ssl#27] [Q&A]
Hey,

I've been running my Foundry instance for a couple of months now and whilst updating a few modules today I had this error pop up and my server just gives me a 504 gateway error now any time I try and get onto it. I've restarted the instance and whilst that lets me load up the webpage, it immediately feeds me the same error and returns to 504-ing whenever I try to do anything. I've had a quick google around as to what might be causing this and whilst I've got a couple ideas I really don't want to wind up messing this up and losing my instance.

**Comment by cat-box** [comment]
First thing I would do, SSH into the ec2 instance if you can, and back up your `/foundrydata` folder. If you've enabled snapshots, you should already have a fairly recent backup.

It sounds like the volume attached to the EC2 instance is full. You'll have to manually expand it through the console, and then SSH into the EC2 instance to expand the file system. AWS has a step by step document [here](https://aws.amazon.com/premiumsupport/knowledge-center/expand-ebs-root-volume-windows/)

Where are you storing your world items? Are you using the S3 bucket from the AWS console?

**Reply by TheWomanInYellow** [reply]
So I expanded the volume and that's got me up and running again, so thanks for that! I use my S3 bucket in terms of whenever I upload anything on my Foundry instance; music, images, etc., etc. In terms of where my worlds are stored however that's where it defaulted to, which seems to be within the EC2 instance. I'm not entirely sure how I'd get that being drawn from my S3 bucket instead?

**Reply by cat-box** [reply]
sounds like you're using it correctly! I just wanted to make sure :) Installed systems do have to be stored in the root volume. 

make sure you're aware of the following: [EBS Pricing](https://aws.amazon.com/ebs/pricing/) and [EBS Bills](https://aws.amazon.com/premiumsupport/knowledge-center/ebs-volume-charges/)
